src/swap.ts

```diff
@@ -256,16 +256,14 @@ function outputFoundInTransaction(
           blindKey
         );
         // check unblind value and unblind asset
-        console.log('unblind val | val', unblindValue, value);
-        console.log('unblind asset | asset', toAssetHash(unblindAsset), asset);
         return unblindValue === value && toAssetHash(unblindAsset) === asset;
       } catch (_) {
         // if unblind fail --> return false
         return false;
       }
     }
     // check value and asset
-    const isAsset: boolean = toAssetHash(o.asset) === asset;
+    const isAsset: boolean = toAssetHash(o.asset).trim() === asset.trim();
```

Comment:
I don't know what happen, maybe I miss something but the asset hash string equality fails for `88fd7f207020e04d7296fb2100505107b2b026e1cc420a5408b3a07f015521ae`. 

I tried to add `trim` to avoid white spaces but the problem persist.
Any idea about that ?

Reply:
Problem solved using `Buffer.equals` function to test asset hash equality.